nexus/db-queries/src/db/datastore/db_metadata.rs

```diff
@@ -213,93 +215,157 @@ impl DataStore {
         info!(log, "Reading schemas from {}", config.schema_dir);
         let mut dir = tokio::fs::read_dir(&config.schema_dir)
             .await
-            .map_err(|e| format!("Failed to read schema config dir: {e}"))?;
+            .context("Failed to read schema config dir")?;
         let mut all_versions = BTreeSet::new();
-        while let Some(entry) = dir
-            .next_entry()
-            .await
-            .map_err(|e| format!("Failed to read schema dir: {e}"))?
+        while let Some(entry) =
+            dir.next_entry().await.context("Failed to read schema dir")?
         {
-            if entry.file_type().await.map_err(|e| e.to_string())?.is_dir() {
+            if entry.file_type().await?.is_dir() {
                 let name = entry
                     .file_name()
                     .into_string()
-                    .map_err(|_| "Non-unicode schema dir".to_string())?;
+                    .map_err(|_| anyhow!("Non-unicode schema dir"))?;
                 if let Ok(observed_version) = name.parse::<SemverVersion>() {
                     all_versions.insert(observed_version);
                 } else {
                     let err_msg =
                         format!("Failed to parse {name} as a semver version");
                     warn!(log, "{err_msg}");
-                    return Err(err_msg);
+                    bail!(err_msg);
                 }
             }
         }
 
         if !all_versions.contains(&current_version) {
-            return Err(format!(
+            bail!(
                 "Current DB version {current_version} was not found in {}",
                 config.schema_dir
-            ));
+            );
         }
         // TODO: Test this?
         if !all_versions.contains(&desired_version) {
-            return Err(format!(
+            bail!(
                 "Target DB version {desired_version} was not found in {}",
                 config.schema_dir
-            ));
+            );
         }
 
         let target_versions = all_versions.range((
             Bound::Excluded(&current_version),
             Bound::Included(&desired_version),
         ));
 
-        for target_version in target_versions.into_iter() {
+        // Iterate over semver versions (e.g., "1.0.0 -> 2.0.0").
```

Comment:
Worth making it clear that we're iterating over real versions here (i.e. directories), not prerelease versions (files)?

Reply:
Update the comments to make this more clear!